Use KUBECTL_PATH if it's defined

gouyang opened the pull request:
It could fix #26869 as a side effect.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/26875)

<!-- Reviewable:end -->

gouyang committed:
Use KUBECTL_PATH if it's defined

It could fix #26869 as a side effect.

zmerlynn commented on cluster/common.sh:
```diff
@@ -258,13 +258,14 @@ function load-or-gen-kube-bearertoken() {
   export KUBECONFIG=${KUBECONFIG:-$DEFAULT_KUBECONFIG}
+  export KUBECTL_PATH=${KUBECTL_PATH:-${KUBE_ROOT}/cluster/kubectl.sh}
 
-  local cc=$("${KUBE_ROOT}/cluster/kubectl.sh" config view -o jsonpath="{.current-context}")
```
This pattern exists all throughout `cluster.sh`, so changing this one place is fairly limited. Please expand this PR if you want to do this.

gouyang committed:
Merge remote-tracking branch 'upstream/master' into improve_cluster_common

gouyang committed:
Use KUBECTL_PATH if it's defined

cluster/kubectl.sh will source cluster/kube-util.sh, which is causing
some problems like #26869, use KUBECTL_PATH is robust.

gouyang commented:
@zmerlynn Updated, please review.

zmerlynn commented:
@k8s-bot test this, issue #IGNORE (missing test results)
